modules/olmv1-creating-a-cluster-role.adoc

```diff
@@ -0,0 +1,334 @@
+
+.Prerequisites
+
+* Access to an {product-title} cluster using an account with `cluster-admin` permissions.
+* You have downloaded the manifests in the image reference of the extension you want to install.
+
+.Procedure
+
+. Create a new cluster role manifest, similar to the following example:
++
+.Example `<extension>-cluster-role.yaml` file
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: <extension>-installer-clusterrole
+----
+
+. Edit your cluster role manifest to include permission to update finalizers on the extension, similar to the following example:
++
+.Example <extension>-cluster-role.yaml
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: pipelines-installer-clusterrole
+rules:
+- apiGroups:
+  - olm.operatorframework.io
+  resources:
+  - clusterextensions/finalizers
+  verbs:
+  - update
+  # Scoped to the name of the ClusterExtension
+  resourceNames:
+  - <metadata_name> # <1>
+----
+<1> Specifies the value from the `metadata.name` field from the custom resource (CR) of the extension.
+
+. Search for the `clusterrole` and `clusterrolebindings` values in the `rules.resources` field in the extension's CSV file. Copy the API groups, resources, verbs, and resource names to your manifest, similar to the following example:
++
+.Example cluster role manifest
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: pipelines-installer-clusterrole
+rules:
+# ...
+# ClusterRoles and ClusterRoleBindings for the controllers of the extension
+- apiGroups:
+  - rbac.authorization.k8s.io
+  resources:
+  - clusterroles
+  verbs:
+  - create # <1>
+  - list
+  - watch
+- apiGroups:
+  - rbac.authorization.k8s.io
+  resources:
+  - clusterroles
+  verbs:
+  - get
+  - update
+  - patch
+  - delete
+  resourceNames: # <2>
+  - "*"
+- apiGroups:
+  - rbac.authorization.k8s.io
+  resources:
+  - clusterrolebindings
+  verbs:
+  - create
+  - list
+  - watch
+- apiGroups:
+  - rbac.authorization.k8s.io
+  resources:
+  - clusterrolebindings
+  verbs:
+  - get
+  - update
+  - patch
+  - delete
+  resourceNames:
+  - "*"
+# ...
+----
+<1> You cannot scope `create`, `list`, and `watch` permissions to specific resource names (the `resourceNames` field). You must scope these permissions to their resources (the `resources` field).
+<2> Some resource names are generated by using the following format: `<package_name>.<hash>`. After you install the extension, look up the resource names for the cluster roles and cluster role bindings for the controller of the extension. Replace the wildcard characters in this example with the generated names and follow the principle of least privilege.
+
+. Search for the `customresourcesdefinition` value in the `rules.resources` field in the extension's CSV file. Copy the API groups, resources, verbs, and resource names to your manifest, similar to the following example:
```

Comment:
```suggestion
. Find the `customresourcesdefinitions` value in the `rules.resources` field in the extension's CSV file, and copy the API groups, resources, verbs, and resource names into your manifest, similar to the following example:
```

Reply:
Maybe something like this works?

Reply:
I went with the subbullet approach. That sentence would be way to long for my liking. I'm already fighting to keep this below an 11th grade reading level. :)